lib/ansible/galaxy/dependency_resolution/providers.py

```diff
@@ -182,6 +181,17 @@ def is_satisfied_by(self, requirement, candidate):
         A boolean should be returned to indicate whether `candidate` is a
         viable solution to the requirement.
         """
+        # NOTE: Only allow pre-release candidates if we want pre-releases or
+        # the req ver was an exact match with the pre-release version.
```

Comment:
Do we want this for the initial direct install requests or transitive dependencies too?